R/assertions.r

```diff
@@ -43,11 +44,11 @@ on_failure(has_attr) <- function(call, env) {
 #' @export
 #' @rdname has_attr
 has_name <- function(x, which){
-    assert_that(is.scalar(which), msg = "multiple values passed to has_name")
-    which %in% names(x)
+    all(which %in% names(x))
 }
 on_failure(has_name) <- function(call, env) {
-  paste0(deparse(call$x), " does not have name ", eval(call$which, env))
+    out_names <- paste0("'", paste0(eval(call$which, env), collapse = "', '"), "'")
+    paste0(deparse(call$x), " does not have all of these name(s): ", out_names)
```

Comment:
because of the possibility that `call$which` is a vector with length greater than 1, I had to do this `paste0()` stuff to avoid getting 

> Error in stop(assertError(attr(res, "msg"))) : bad error message